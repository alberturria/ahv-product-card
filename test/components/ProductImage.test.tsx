import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductTitle', () => {
  test('Should have render the component properly with the image', () => {
    const wrapper = renderer.create(
      <ProductImage img="randomImage.png" className="custom-class" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Should render the component with the name of the product', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
