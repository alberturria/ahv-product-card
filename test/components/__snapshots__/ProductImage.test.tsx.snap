// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Should have render the component properly with the image 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="randomImage.png"
/>
`;

exports[`ProductTitle Should render the component with the name of the product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
